dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -1593,7 +1604,7 @@ def _get_max(
         subset_properties: dict,
     ) -> None:
         max_value = df_series.max()
-        self.max = max_value if not self.max else max(self.max, max_value)
+        self.max = float(max_value) if not self.max else float(max(self.max, max_value))
```

Comment:
Cast added to adhere to type specification in numeric stats mixin attribute initialization (avoids setting to np.float64)